qa_config/pmd/ruleset.xml

```diff
@@ -22,8 +22,9 @@
     <!--TODO evaluate this rule again. See class OnFailureDestinationEvent as example of why we disabled it -->
     <exclude name="GenericsNaming"/>
     <exclude name="ShortClassName"/>
-
+    <exclude name ="UseExplicitTypes"/>
     <exclude name="EmptyMethodInAbstractClassShouldBeAbstract"/>
+    <exclude name="UnnecessaryCast"/>
```

Comment:
PMD complains about casting ints to doubles but I am not sure about the correctness of the result